test/core/end2end/tests/default_host.c

```diff
@@ -201,7 +201,7 @@ static void simple_request_body(grpc_end2end_test_fixture f) {
   GPR_ASSERT(status == GRPC_STATUS_UNIMPLEMENTED);
   GPR_ASSERT(0 == strcmp(details, "xyz"));
   GPR_ASSERT(0 == strcmp(call_details.method, "/foo"));
-  GPR_ASSERT(0 == strcmp(call_details.host, "localhost"));
+  GPR_ASSERT(0 == strncmp(call_details.host, "localhost", 9));
```

Comment:
isn't strncmp equivalent to strcmp in this case, given that one of the arguments is a string literal?

In any case, if we wanted to use the (arguably better) strncmp flavor, we should update the other calls right above for consistency IMHO.